src/main/java/io/openepi/weather/api/WeatherApi.java

```diff
@@ -0,0 +1,380 @@
+        <tr><td> Status Code </td><td> Description </td><td> Response Headers </td></tr>
+        <tr><td> 200 </td><td> Successful Response </td><td>  -  </td></tr>
+        <tr><td> 422 </td><td> Validation Error </td><td>  -  </td></tr>
+     </table>
+     */
+    public METJSONSunrise getSunriseSunriseGet(BigDecimal lat, BigDecimal lon, String date) throws ApiException {
+        ApiResponse<METJSONSunrise> localVarResp = getSunriseSunriseGetWithHttpInfo(lat, lon, date);
+        return localVarResp.getData();
+    }
+
+    /**
+     * Get sunrise and sunset information
+     * Returns the sunrise time and sunset time for the given location
+     * @param lat Latitude (required)
+     * @param lon Longitude (required)
+     * @param date Date (optional)
+     * @return ApiResponse&lt;METJSONSunrise&gt;
+     * @throws ApiException If fail to call the API, e.g. server error or cannot deserialize the response body
+     * @http.response.details
+     <table summary="Response Details" border="1">
+        <tr><td> Status Code </td><td> Description </td><td> Response Headers </td></tr>
+        <tr><td> 200 </td><td> Successful Response </td><td>  -  </td></tr>
+        <tr><td> 422 </td><td> Validation Error </td><td>  -  </td></tr>
+     </table>
+     */
+    public ApiResponse<METJSONSunrise> getSunriseSunriseGetWithHttpInfo(BigDecimal lat, BigDecimal lon, String date) throws ApiException {
+        okhttp3.Call localVarCall = getSunriseSunriseGetValidateBeforeCall(lat, lon, date, null);
+        Type localVarReturnType = new TypeToken<METJSONSunrise>(){}.getType();
+        return localVarApiClient.execute(localVarCall, localVarReturnType);
+    }
+
+    /**
+     * Get sunrise and sunset information (asynchronously)
+     * Returns the sunrise time and sunset time for the given location
+     * @param lat Latitude (required)
+     * @param lon Longitude (required)
+     * @param date Date (optional)
+     * @param _callback The callback to be executed when the API call finishes
+     * @return The request call
+     * @throws ApiException If fail to process the API call, e.g. serializing the request body object
+     * @http.response.details
+     <table summary="Response Details" border="1">
+        <tr><td> Status Code </td><td> Description </td><td> Response Headers </td></tr>
+        <tr><td> 200 </td><td> Successful Response </td><td>  -  </td></tr>
+        <tr><td> 422 </td><td> Validation Error </td><td>  -  </td></tr>
+     </table>
+     */
+    public okhttp3.Call getSunriseSunriseGetAsync(BigDecimal lat, BigDecimal lon, String date, final ApiCallback<METJSONSunrise> _callback) throws ApiException {
+
+        okhttp3.Call localVarCall = getSunriseSunriseGetValidateBeforeCall(lat, lon, date, _callback);
+        Type localVarReturnType = new TypeToken<METJSONSunrise>(){}.getType();
+        localVarApiClient.executeAsync(localVarCall, localVarReturnType, _callback);
+        return localVarCall;
+    }
+}
```

Comment:
As a general note, is it possible to make the public interface of this (and the other clients) easier for end users?

Reply:
Yes, a lot can be improved.

I believe there are several improvements that can be done. However, each change we do strays further away from the generated code and makes it harder to just "copy" it if we make a change to the APIs later. In other words, more to maintain. But I definitely see the need to make them "simpler". I think it should be a separate PR though.

Reply:
Agree to do this in a separate PR. 
The reason for doing this is that I think that a designated client library should create more value for the user, than just allowing the user to skip doing the client generation themselves.

Reply:
Agree on that 😄